apps/crn-frontend/src/analytics/collaboration/Collaboration.tsx

```diff
@@ -45,14 +63,38 @@ const Collaboration = () => {
         .collaboration({})
         .collaborationPath({ metric: newMetric, type }).$,
     );
-  const setType = (newType: 'within-team' | 'across-teams') =>
+  const setType = (newType: 'within-team' | 'across-teams') => {
+    if (metric === 'user') {
+      setUserSort('user_asc');
+      setUserSortingDirection(userCollaborationInitialSortingDirection);
+    } else {
+      setTeamSort('team_asc');
+      setTeamSortingDirection(teamCollaborationInitialSortingDirection);
+    }
     history.push(
       analytics({})
         .collaboration({})
         .collaborationPath({ metric, type: newType }).$,
     );
+  };
+
+  const { client: userClient } = useAnalyticsUserCollaboration({
```

Comment:
I was wondering if we could get this client by a function like

```
  const getUserClient = (sort: SortUserCollaboration) => {

    const indexName =
    sort === 'user_asc'
    ? ANALYTICS_ALGOLIA_INDEX
    : `${ANALYTICS_ALGOLIA_INDEX}_user_${sort.replace('user_', '')}`;
    
    return useAnalyticsAlgolia(indexName);
  } 
```

That would be called here and inside `useAnalyticsUserCollaboration` instead of making
`useAnalyticsUserCollaboration` to return `userClient`. Because the way it is I believe it's doing more calls than it need to.

Here on https://4342.hub.asap.science/analytics/collaboration/user/within-team after cleaning the network and refreshing the page, we have 14 requests

<img width="2273" alt="Screenshot 2024-08-02 at 07 48 52" src="https://github.com/user-attachments/assets/56aa4aeb-cb00-451d-9077-d97f03d12f5c">

and in my local without calling `useAnalyticsUserCollaboration` and `useAnalyticsTeamCollaboration` on `apps/crn-frontend/src/analytics/collaboration/Collaboration.tsx` we have 7 requests
<img width="2267" alt="Screenshot 2024-08-02 at 07 49 23" src="https://github.com/user-attachments/assets/1d2e603e-2a7e-4cb8-8748-d45826b61c23">